cmd/progressbar/progressbar.go

```diff
@@ -138,11 +158,17 @@ func (pb *progressBar) handleDynamicBarUpdate(data DynamicBarUpdate) {
 		log.Panicf("trying to update a bar that doesn't exist: %s (%v)", displayName, data)
 	}
 
+	currentTotal := val.Current()
+
 	if data.TotalChange > 0 {
-		val.SetTotal(val.Current()+data.TotalChange, false)
+		val.SetTotal(currentTotal+data.TotalChange, false)
 	}
 
 	if data.Change > 0 {
+		if currentTotal == 0 {
+			val.SetTotal(currentTotal+int64(data.Change), false)
+		}
```

Comment:
either you want it to be `currentTotal > 0`, or the `currentTotal+` is redundant because it equals 0 (actually, I don't quite understand what this line does)

Reply:
wait, if you only update the total once, doesn't it mean that you'll reach bar-complete after the next change? i.e. leading the progress-bar wait to finish?